[bugfix] I2C will now throw an error if bus fails to open

brianjjones opened the pull request:
This is a fix for issue #551, now an error will be thrown if the
I2C bus fails to open.

grgustaf commented on src/zjs_i2c_ipm.c:
```diff
@@ -294,20 +294,24 @@ static jerry_value_t zjs_i2c_open(const jerry_value_t function_obj,
+        zjs_obj_add_function(i2c_obj, zjs_i2c_abort, "abort");
+        zjs_obj_add_function(i2c_obj, zjs_i2c_close, "close");
+        zjs_obj_add_number(i2c_obj, bus, "bus");
+        zjs_obj_add_number(i2c_obj, speed, "speed");
```
This should use a prototype since there can be more than one of these i2c_obj's created.

See how prototype was added to grove_lcd for example with this command:
`git show 81b7fcb src/zjs_grove_lcd.c src/zjs_modules.c`

(In zjs_modules.c, just pay attention where the cleanup function was added.)

Otherwise, LGTM.

grgustaf commented:
@brianjjones: Oh we probably crossed the streams when you were working on this and I was doing the prototype stuff. So make sure you fix that in both the IPM and non-IPM versions if need be.

brianjjones commented:
@grgustaf I've added the fixes needed for both versions to use the prototype, as well as a couple other minor fixes.  It should be good to go now, i've thoroughly tested on both platforms.  It will requires a couple of merge fixes, I can't do this as I don't have write access.  Please note that in arc/src/main.c, for the ERR_PRINT msg_bus is an uint not a string, so use my version rather than master.  And in src/zjs_i2c.c, keep the return zjs_error I added to line 284.  If you have any questions let me know.

grgustaf commented:
+1
